### PR TITLE
Create sig_perm_test.py

### PR DESCRIPTION
**Pull request by granitz**

### CODE REVIEW

**Comment by wiheto** [comment]
I need to spend some time looking through this, unless you would like to run a test or two. Mainly check that the modularity function you have written equals networkx's modularity. I'll get round to this sometime next week if you don't want to do it.

**Comment by granitz** [comment]
> I need to spend some time looking through this, unless you would like to run a test or two. Mainly check that the modularity function you have written equals networkx's modularity. I'll get round to this sometime next week if you don't want to do it.

It returns the same result as the corresponding matlab function in http://commdetect.weebly.com. I'm trying to create some explicit test for it though. Let me know if you come up with something.

File: teneto/sig_perm_test.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    N = np.shape(net)[0] # number of nodes
+
+    cl_label = np.unique(community)
+    C = len(cl_label) # possibly ravel to count all
+    cl = np.zeros((N, C))
+
+    for i in range(C):
+        cl[:, i] = community==cl_label[i]
+
+    """
```
Comment (wiheto) [inline teneto/sig_perm_test.py]: I'm going to delete part of this note. As the tests should answer if you are doing it correctly.

File: teneto/sig_perm_test.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import numpy as np
+import random
```
Comment (wiheto) [inline teneto/sig_perm_test.py]: random is never used so I'll be deleting this

File: teneto/sig_perm_test.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        for j in range(C):
+            tmp[i,j] = np.dot(net[i,:],cl[:,j])
+
+    ct = np.transpose(cl)
+
+    for i in range(C):
+        for j in range(C):
+            Q[i,j] = np.dot(tmp[:,i],ct[j,:])
+
+    return Q
```
Comment (wiheto) [inline teneto/sig_perm_test.py]: Q has a trailing whitespce, will be deleting

**Comment by wiheto** [comment]
So this can be merged once the test issues have been solved.

**Pull request closed without merging**